### PR TITLE
[Android] Fix ItemSpacing for last item in CollectionView

### LINKED ISSUE #7357: [Android] Setting ItemSpacing creates spacing for the last item in CollectionView
### Steps to Reproduce

1. Run the repro.
2. Notice that `ItemSpacing` is set to 5.
3. Observe that the last item (item 49) has bottom spacing.

### Expected Behavior

On Android, `ItemSpacing` seems to be added to the bottom of each item, but it should truly be spacing between items. iOS doesn't have this issue.

_You should also test this with the grid layout. I haven't tested it, but we want to ensure horizontal and vertical item spacing work for all items in the last row (or column) of a grid collection._

### Actual Behavior

There is spacing below the last item.

### Basic Information

- Version with issue: 4.3.0.1106-nightly
- IDE: VS2019
- Affected Devices: Samsung S9

### Reproduction Link

[App105.zip](https://github.com/xamarin/Xamarin.Forms/files/3567262/App105.zip)

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change ###

`ItemSpacing` for the last item in CollectionView should work now. I added a manual test with instructions.

@hartez I noticed that setting `ItemsSpacing` for `LinearItemsLayout` in XAML did not work, but it seems to be working for `ListItemsLayout` in an earlier Nuget. See #7519.

### Issues Resolved ### 

- fixes #7357

### PR Checklist ###

- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jsuarezruiz** [comment]
@adrianknight89 Could you rebase 4.3.0 branch (there is a small conflict)?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems [hunk 1]
```diff
@@ -1355,6 +1358,9 @@
     <Compile Update="$(MSBuildThisFileDirectory)Issue6254.xaml.cs">
       <DependentUpon>Issue6254.xaml</DependentUpon>
     </Compile>
+    <Compile Update="C:\Users\def\Desktop\x\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\Issue7357.xaml.cs">
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems]:
```suggestion
    <Compile Update="$(MSBuildThisFileDirectory)Issue7357.xaml.cs">
```

Comment (adrianknight89) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Xamarin.Forms.Controls.Issues.Shared.projitems]: Thanks. I saw it recently as well.

**Comment by adrianknight89** [comment]
@jsuarezruiz Done.

**Comment by jfversluis** [comment]
Failing tests are unrelated. Mostly the CarouselView ones and one for Issue5766 which does nothing with CollectionView. For the latter verified locally that it does work. Seems to be a hiccup.

**Pull request merged**